File: leetcode/leet_028.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if not needle:
            return 0
        haystack, needle = haystack + "\0", needle + "\0"
        i, L = 0, len(haystack)
        while haystack[i] != "\0":
            if haystack[i] == needle[0]:
                start, now = i, 0
                while haystack[i] == needle[now] and haystack[i] != "\0" and needle[now] != "\0":
                    i += 1
                    start += 1
                else:
                    if needle[now] == "\0":
                        return start
            else:
                i += 1
        return -1

File: leetcode/leet_028.cpp
#include <bits/stdc++.h>

using namespace std;
typedef long long int ll;
typedef pair<int, int> P;
typedef pair<ll, ll> Pll;
typedef vector<int> Vi;
//typedef tuple<int, int, int> T;
#define FOR(i,s,x) for(int i=s;i<(int)(x);i++)
#define REP(i,x) FOR(i,0,x)
#define ALL(c) c.begin(), c.end()
#define DUMP( x ) cerr << #x << " = " << ( x ) << endl

const int dr[4] = {-1, 0, 1, 0};
const int dc[4] = {0, 1, 0, -1};


class Solution {
public:
  int strStr(string haystack, string needle) {
    if (needle.size() == 0) {
      return 0;
    } else if (haystack.size() == 0) {
      return -1;
    }
    haystack += '\0';
    needle += '\0';
    for (int i = 0; ; i++) {
      for (int j = 0; ; j++) {
        if (needle[i] == '\0') return i;
        if (haystack[i + j] == '\0') return -1;
        if (haystack[i + j] != needle[j]) break;
      }
    }
  }
};

int main() {
  // use scanf in CodeForces!
  cin.tie(0);
  ios_base::sync_with_stdio(false);

  return 0;
}
